daemon/daemon.go

```diff
@@ -445,6 +445,18 @@ func (daemon *Daemon) DaemonLeavesCluster() {
 	// Daemon is in charge of removing the attachable networks with
 	// connected containers when the node leaves the swarm
 	daemon.clearAttachableNetworks()
+	// Daemon is in charge of removing the ingress network when the
+	// node leaves the swarm. Wait for job to be done or timeout.
+	// This is called also on graceful daemon shutdown. We need to
+	// wait, because the ingress release has to happen before the
+	// network controller is stopped.
+	done, _ := daemon.ReleaseIngress()
```

Comment:
Let's either check the error here, or change `ReleaseIngress` so it does not return an error (currently it always returns `nil`).